src/pages/public_registration/log/LogMessageAccordion.tsx

```diff
@@ -37,17 +50,37 @@ export const LogMessageAccordion = ({ messages, messageBackgroundColor }: LogMes
             p: 0,
             gap: '0.25rem',
           }}>
-          {messages.map((message) => (
-            <MessageItem
-              key={message.identifier}
-              text={message.text}
-              date={message.createdDate}
-              username={message.sender}
-              backgroundColor={messageBackgroundColor}
-            />
-          ))}
+          {messages.map((message) => {
+            const messageType = getTicketTypeFromLogEntryTopic(topic);
+            const canDeleteMessage = !!user && userCanDeleteMessage(user, message, messageType);
+            return (
+              <MessageItem
+                key={message.identifier}
+                text={message.text}
+                date={message.createdDate}
+                username={message.sender}
+                backgroundColor={messageBackgroundColor}
+                menuElement={
+                  canDeleteMessage && (
+                    <MessageMenu messageId={message.id} refetchData={refetchData} disableDelete={!!message.text} />
+                  )
+                }
+              />
+            );
+          })}
         </Box>
       </Collapse>
     </Box>
   );
 };
+
+const getTicketTypeFromLogEntryTopic = (topic: LogEntry['topic']): TicketType => {
```

Comment:
Velkommen i klubben med å ha en switch her man må vedlikeholde :D 
Om det kommer inn `DoiReserved` eller `NviApproval` eller what not i fremtiden vil vi måtte gå gjennom alle og oppdatere, for nå vil de to eksemplene automatisk bli `PublishingRequest` her i fremtiden 😬 Verste av alt er at vi vil ikke få noen indikasjon på det før noen merker at noe er feil. Men fant ikke noen bedre løsning på det i alle de der som ligger i `LogEntryItem`, så da får det bare være sånn til noen finner en bedre løsning 🤷‍♂️